boray/src/main.rs

```diff
@@ -78,31 +69,36 @@ fn main() {
     let pg_config = PostgresConnectionConfig {
         user: Some(config.database.user),
         password: None,
-        host: Some(config.database.host),
-        port: Some(config.database.port),
+        host: None,
+        port: None,
         database: Some(config.database.database),
         application_name: Some(config.database.application_name),
         tls_config,
     };
 
     let connection_creator = PostgresConnection::connection_creator(pg_config);
 
-    // When the manatee resolver is complete then this will need to log the
-    // dynamic backend in some other way.  For now, at least reporting backend
-    // ip/port will be helpful.
+    //
+    // TODO log the dynamic backend IP somehow? The resolver will at least emit
+    // log entries when the IP changes.
+    //
     let pool_opts = ConnectionPoolOptions {
-        maximum: config.cueball.max_connections,
+        max_connections: Some(config.cueball.max_connections),
         claim_timeout: config.cueball.claim_timeout,
-        log: log.new(o!(
-            "component" => "CueballConnectionPool",
-            "backend_ip" => pg_ip.to_string(),
-            "backend_port" => config.database.port,
-        )),
+        log: Some(log.new(o!(
+            "component" => "CueballConnectionPool"
+        ))),
         rebalancer_action_delay: config.cueball.rebalancer_action_delay,
+        decoherence_interval: None
     };
 
-    let primary_backend = (pg_ip, config.database.port);
-    let resolver = StaticIpResolver::new(vec![primary_backend]);
+    let resolver = ManateePrimaryResolver::new(
+        config.zookeeper.zk_conn_str.parse::<ZkConnectString>().unwrap(),
```

Comment:
See my other comment about making the connection string type `ZkConnectString`.

Reply:
Done